spock-core/src/main/java/org/spockframework/util/ExceptionUtil.java

```diff
@@ -24,9 +24,20 @@
 public class ExceptionUtil {
   /**
    * Allows to throw an unchecked exception without declaring it in a throws clause.
+   * The given {@code Throwable} will be unconditionally thrown.
+   * The return type is only present to be able to use the method as
+   * last statement in a method or lambda that needs a value returned,
+   * but the method will never return successfully.
+   *
+   * @param throwable the throwable to be thrown
+   * @param <R>       the fake return type
+   * @param <T>       the class of the throwable that will be thrown
+   * @return nothing as the method will never return successfully
+   * @throws T unconditionally
    */
-  public static void sneakyThrow(Throwable t) {
-    ExceptionUtil.<RuntimeException>doSneakyThrow(t);
+  @SuppressWarnings("unchecked")
+  public static <R, T extends Throwable> R sneakyThrow(Throwable t) throws T {
```

Comment:
@Vampire do we really need a fake return? Some code quality rules complain about ignoring the return value.

Reply:
Did you see such a complaint with that method?
I only know of tools that complain about specific methods where the return value usually should be considered and it might be an error to ignore it, but not for all return values.
There are many methods where the return value can or can not be interesting and it is ok to ignore it.

_You_ probably don't **need** it, but I needed a `sneakyThrow` too and I needed a return value for easier usage like being able to do.
```java
completableFuture
    .handle((user, throwable) -> sneakyThrow(throwable));
```
instead of 
```java
completableFuture
    .handle((user, throwable) -> {
        sneakyThrow(throwable);
        return null;
    });
```
where the `return` is just unnecessary boilerplate as it cannot be reached ever anyway but the compiler cannot see that.

Reply:
I agree that (in Java) with a lambda expression throwing a checked exception it is really useful. Some time ago I also added similar construction in my [mockito-java8](https://github.com/szpak/mockito-java8/blob/master/src/main/java/info/solidsoft/mockito/java8/SneakyThrow.java) code (which is heavily used in companion with lambdas) , based on the user request to "do something with those ugly checked exceptions in lambdas which make code longer and less readable " .

Reply:
I now also updated the usages where a fake `return null;` was done after the `sneakyThrow` call to leverage the return type.

Reply:
I now also updated the usages where a fake `return null;` was done after the `sneakyThrow` call to leverage the return type.